src/kontrol/__main__.py

```diff
@@ -375,15 +376,30 @@ def exec_init(options: InitOptions) -> None:
 
 
 # Helpers
-def _loglevel(args: Namespace) -> int:
-    if hasattr(args, 'debug') and args.debug:
+def _loglevel(args: Namespace, toml_args: dict) -> int:
+    if (not hasattr(args, 'debug') or args.debug is None) and ('debug' in toml_args and toml_args['debug']):
+        return logging.DEBUG
+    elif hasattr(args, 'debug') and args.debug:
         return logging.DEBUG
 
-    if hasattr(args, 'verbose') and args.verbose:
+    if (not hasattr(args, 'verbose') or args.verbose is None) and ('verbose' in toml_args and toml_args['verbose']):
+        return logging.INFO
+    elif hasattr(args, 'verbose') and args.verbose:
         return logging.INFO
 
     return logging.WARNING
```

Comment:
You can extract this logic into an auxiliary function to improve readability. Also, maybe using `getattr` instead of `hasattr`. Something like
```py
def _loglevel(args: Namespace, toml_args: dict) -> int:
    def is_attr_used(attr_name: str) -> bool:
        return getattr(args, attr_name, None) is not None or toml_args.get(attr_name)

    if is_attr_used('debug'):
        return logging.DEBUG

    if is_attr_used('verbose'):
        return logging.INFO

    return logging.WARNING
```